packages/SwingSet/src/controller/controller.js

```diff
@@ -498,6 +500,27 @@ export async function makeSwingsetController(
       }
       return kpid;
     },
+
+    upgradeStaticVat(vatName, shouldPauseFirst, bundleID, options = {}) {
+      const vatID = kernel.vatNameToID(vatName);
+      let readyKpidBody = null;
+      let readyKpidSlots = [];
+      if (shouldPauseFirst) {
+        const readyKpid = controller.queueToVatRoot(vatName, 'pauseService');
+        readyKpidBody = { [QCLASS]: 'slot', index: 0 };
+        readyKpidSlots = [readyKpid];
+      }
+      if (!options.upgradeMessage) {
+        options.upgradeMessage = `vat ${vatName} upgraded`;
+      }
+      return controller.queueToVatRoot(
+        vatName,
+        'upgradeVat',
```

Comment:
Exposing an `upgradeStaticVat` method on the vat admin root (to be implemented in terms of the existing `upgradeVat` function that would be made internal) would enable expressing the `upgradeStaticVat` method in terms of the vat name rather than the vat ID, which I think would be an improvement as it would move the name lookup into the vat admin vat.  It also strikes me that the `await` for the optional pause could be more elegantly handled that way because the `await` could go before the invocation of `upgradeVat` rather than having to be awkwardly embedded inside it.